src/models/event.js

```diff
@@ -340,19 +340,34 @@ utils.extend(module.exports.MatrixEvent.prototype, {
      * @internal
      *
      * @param {module:crypto} crypto crypto module
+     * @param {boolean} cancelAndRetry retry the attempt, even if we tried to decrypt
+     *                                 this event before. If there's an existing
+     *                                 pending key request for this event, cancel it
+     *                                 before retrying.
      *
      * @returns {Promise} promise which resolves (to undefined) when the decryption
      * attempt is completed.
      */
-    attemptDecryption: async function(crypto) {
+    attemptDecryption: async function(crypto, cancelAndRetry=false) {
         // start with a couple of sanity checks.
         if (!this.isEncrypted()) {
             throw new Error("Attempt to decrypt event which isn't encrypted");
         }
 
-        if (
-            this._clearEvent && this._clearEvent.content &&
+        if (cancelAndRetry) {
```

Comment:
the comment says "start with a couple of sanity checks." This isn't a sanity check. Again, please try not to make comments lie.